### PR TITLE
fix catching unusual method declarations

### LINKED ISSUE #24: Arguments are not well caught
When arguments are being added in an unusual format.
Like:
```python
def hello(x=1,
               y=2,
               z=3)
```

Output

Screen Shot 2019-05-09 at 4 59 53 PM

### PR DESCRIPTION
**Pull request by alucardthefish**
Adds flexibility for methods that are declared using newlines in the parameter section.
Now it works as normal as if it was declared in a single line. Fixes #24

### CODE REVIEW

**Comment by alucardthefish** [comment]
> I don't think that's working.
> 
> Try `dyc diff` on this
> 
> ```python
> def hello_world(x=1,
>           y=2,
>           z=3):
>     return 'Test'
> ```
> 
> It doesn't trigger to document.

This bug fixing was intended for:

`dyc start` command as the output in the issue #24 indicates:
image

I would like to see your output for `dyc diff` anyways. I've tried to check its functionality on master branch but I couldn't replicate the behavior explained on usage and examples section of the repo. I think this problem in diff command should be another issue.

**Comment by Zarad1993** [comment]
@alucardthefish - I think the issue right here is with the commits you have. It makes diff not work. For example, before the commits here. `dyc diff` let's you catch the the method above _although not accurate_. But after applying your branch. It ignores the whole method.

Another question, I'm not sure I see where the issue #24 indicates that it's for `start` and not `diff` ? The issue is badly written by me without accurately indicating which to use. Nonetheless, a fix for one shouldn't be breaking the other

What do you think?

**Comment by alucardthefish** [comment]
> @alucardthefish - I think the issue right here is with the commits you have. It makes diff not work. For example, before the commits here. `dyc diff` let's you catch the the method above _although not accurate_. But after applying your branch. It ignores the whole method.
> 
> Another question, I'm not sure I see where the issue #24 indicates that it's for `start` and not `diff` ? The issue is badly written by me without accurately indicating which to use. Nonetheless, a fix for one shouldn't be breaking the other
> 
> What do you think?

@Zarad1993  Issue #24 indicated it indirectly. That's why I copied the image you posted in which shows the behavior of `dyc start` command. I agree that the issue was badly written though. However, if you say that it crashes that part of `dyc diff` I will check that out. The bad thing is that I don't know how it works since in master branch I tried to replicate the usage and doesn't show anything. I locate myself in the initialize directory and try to make changes to an unstaged file as example section says but nothing happen. That's maybe why I didn't test the fixing in that part. I will try to fix the collateral damage made by this fixing. If not just ignore this PR.

**Comment by alucardthefish** [comment]
> I don't think that's working.
> 
> Try `dyc diff` on this
> 
> ```python
> def hello_world(x=1,
>           y=2,
>           z=3):
>     return 'Test'
> ```
> 
> It doesn't trigger to document.

Sorry, I was testing all the time with `dyc diff --watch` and apparently I can't figure out how it works, my bad. Due to the fact that I haven't tested it I didn't know `dyc diff` has similar behavior as `dyc start`.

The weird thing is that it's working for me. Here some snapshots for the example you gave me to try:

console

The output:

Screenshot from 2019-10-12 16-04-37

**Comment by Zarad1993** [comment]
Nice!! I'll test those and get back to you! Really appreciate your follow up 🙂

**Comment by Zarad1993** [comment]
@alucardthefish - Hey man, unfortunately the test is still failing on `dyc diff`.

Check this GIF -> https://recordit.co/XkOt4Pz12k

**Comment by alucardthefish** [comment]
> @alucardthefish - Hey man, unfortunately the test is still failing on `dyc diff`.
> 
> Check this GIF -> https://recordit.co/XkOt4Pz12k

Hey @Zarad1993  I checked the GIF and that was strange. BTW I made some changes and squash them to the PR. Please check if now it's working. If not let me know.

**Comment by alucardthefish** [comment]
Hey @Zarad1993 did it work or not? Remember to let me know please.

**Comment by Zarad1993** [comment]
Hey @alucardthefish - Circling back, this seems to work great on new files. However, when editing existing files it doesn't seem to catch it with `git diff`.. I am reviewing the code to see if this could be addressed in a follow up PR before another release.

File: dyc/base.py [hunk 1]
```diff
@@ -22,22 +22,34 @@ def initialize(self, change=None):
...
-                        word.lstrip()
-                        for word in line.split(' ')
-                        if word.lstrip() in keywords
-                    ]
-                )
-                > 0
-            )
+            keywords = self.config.get("keywords")
+            foundList = [
+                word.lstrip() for word in line.split(" ") if word.lstrip() in keywords
```
Comment (Zarad1993) [inline dyc/base.py:35]: can `is_one_line_method` be used here as well?

**Comment by Zarad1993** [comment]
Hey @alucardthefish - Circling back, this seems to work great on new files. However, when editing existing files it doesn't seem to catch it with `git diff`.. I am reviewing the code to see if this could be addressed in a follow-up PR before another release.

start argument command seems to be working well too

**Comment by Zarad1993** [comment]
Alright I found the issue with `_is_line_part_of_patches` not catching the diff. I will open a follow up PR to fix this.

Nice work on this!!

**Comment by alucardthefish** [comment]
Hey @Zarad1993 Nice to hear from this project.

It was a pleasure to contribute to this project.

Take care!

**Pull request merged**